Run github windows tests on PR push

rvl opened the pull request:
### Issue Number

#2189 

### Overview

- Run windows tests when PR branches are pushed.
- Run windows tests when the master branch or bors branches are updated.
- Don't run windows tests on a nightly schedule any more.
- Some nightly test failures were because of a ssh issue on github actions. So added a fix for the branch update script.
- Integration tests are run on bors branches.

### Comments

- `cardano-wallet-core:unit` is failing due to #2526, so it has `continue-on-error: true`
- `cardano-wallet-launcher:unit` is failing, but meh, so it has `continue-on-error: true`.

rvl committed:
Run github windows tests on PR push

rvl committed:
Fix .buildkite/push-branch.sh script for github actions

It was failing ssh auth due to the host key check.

KtorZ commented on .github/workflows/ci.yml:
```diff
@@ -1,21 +1,32 @@
+      - bors/staging
+      - bors/trying
+  pull_request:
+    types: [opened, synchronize, reopened]
```
| type | semantic |
| --- | --- | 
| opened | pull request is created |
| synchronize | commit(s) pushed to the pull request |
| reopened | closed pull request is reopened | 

(source: https://frontside.com/blog/2020-05-26-github-actions-pull_request/#how-do-workflows-trigger-on-pull_request)

Minor remark: maybe 'reopened' isn't needed here :thinking: ? If it's closed, it also means that tests were run previously already.

piotr-iohk replied:
From the same source :point_up: :
_When a Github Actions workflow is configured to run on pull requests, by default it will be triggered on three different types of activities: opened, synchronize, and reopened._

```
name: Pull Request Workflow
on: pull_request
```
I think default is fine, and no need to specify types. :man_shrugging:

rvl replied:
Yes you're right, I was just getting very confused by the docs.

It does seem like the 'reopened' event is not useful for this kind of workflow, but I will use the default to keep it simpler.

KtorZ commented on .github/workflows/ci.yml:
```diff
@@ -32,7 +43,7 @@ jobs:
         with:
           name: cardano-wallet-tests-win64
-      - run: '.\\cardano-wallet-core-test-unit.exe --color'
+      - run: '.\\cardano-wallet-core-test-unit.exe --color +RTS -N1'
```
:crossed_fingers: didn't help locally for me. Let's see here :man_shrugging:

rvl replied:
OK https://github.com/input-output-hk/cardano-wallet/pull/2527#issuecomment-780439348 I believe you :+1:

I will change this to `-N2` to override the `ghc -with-rtsopts=-N4` setting from cabal. Then it should set the number of threads to match [the VM specs](https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources).

(edit: `-N` by itself did not seem to work)

KtorZ replied:
Sounds like getting rid of parallel didn't help :( 
We may have some property tests that are a bit too much for Windows?

rvl replied:
Something odd is going on with RTS opts. It's supposed to have `-M2G` set for both windows and linux.

KtorZ replied:
Something I've experienced actually with specifying rts opts directly in the cabal file is that:

```
      "-with-rtsopts=-M2G"
      "-with-rtsopts=-N4"
```

This doesn't work. Only the last option gets taken into account. But this does:

```
      - '"-with-rtsopts=-M2G -N4"'
```

rvl committed:
Make checks names shorter

rvl committed:
Skip Cardano.Wallet.DB.Sqlite tests on windows

Because they seem to run out of memory.

  Failures:

    test/unit/Cardano/Wallet/DB/SqliteSpec.hs:416:13:
    1) Cardano.Wallet.DB.Sqlite, Sqlite State machine tests, Sequential
         uncaught exception: SqliteException
         SQLite3 returned ErrorNoMemory while attempting to perform step: out of memory
         (after 740 tests)

    To rerun use: --match "/Cardano.Wallet.DB.Sqlite/Sqlite State machine tests/Sequential/"

    test/unit/Cardano/Wallet/DB/SqliteSpec.hs:424:13:
    2) Cardano.Wallet.DB.Sqlite, Sqlite State machine (RndState), Sequential state machine tests
         uncaught exception: SqliteException
         SQLite3 returned ErrorNoMemory while attempting to perform step: out of memory
         (after 8 tests)

    To rerun use: --match "/Cardano.Wallet.DB.Sqlite/Sqlite State machine (RndState)/Sequential state machine tests/"

    test/unit/Cardano/Wallet/DB/SqliteSpec.hs:668:9:
    3) Cardano.Wallet.DB.Sqlite, Sqlite query logging, should log queries at DEBUG level
         uncaught exception: SqliteException
         SQLite3 returned ErrorNoMemory while attempting to perform step: out of memory

    To rerun use: --match "/Cardano.Wallet.DB.Sqlite/Sqlite query logging/should log queries at DEBUG level/"

  Randomized with seed 1032365168

  Finished in 757.1717 seconds
  2946 examples, 3 failures, 4 pending
